sdk/core/azure-core/src/test/java/com/azure/android/core/internal/util/serializer/DateTimeRfc1123SerializerTest.java

```diff
@@ -0,0 +1,46 @@
+package com.azure.android.core.internal.util.serializer;
+
+import com.azure.android.core.util.DateTimeRfc1123;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+
+import org.junit.Test;
+import org.threeten.bp.LocalDateTime;
+import org.threeten.bp.OffsetDateTime;
+import org.threeten.bp.ZoneId;
+import org.threeten.bp.ZoneOffset;
+import org.threeten.bp.format.DateTimeFormatter;
+
+import java.io.IOException;
+import java.util.Locale;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class DateTimeRfc1123SerializerTest {
+    public static final DateTimeFormatter RFC1123_DATE_TIME_FORMATTER =
+        DateTimeFormatter.ofPattern("EEE, dd MMM yyyy HH:mm:ss 'GMT'").withZone(ZoneId.of("UTC")).withLocale(Locale.US);
+
+    @Test
+    public void test_getModule() {
+        SimpleModule module = DateTimeRfc1123Serializer.getModule();
+
+        assertNotNull(module);
+    }
+
+    @Test
+    public void serializeDateTimeRfc1123() throws IOException {
```

Comment:
Mind adding a test for non-DateTimeRfc1123 or malformed representations being passed into this serializer? Would it be able to handle an ISO8601 date? What happens if `GMT` is missing from the date string? Could it handle a different time zone offset?

Reply:
From what I've noticed playing around with this serializer it only accepts `DateTimeRfc1123` objects. To create an instance of this class you need either an `OffsetDateTime` instance or a correctly formatted string, which make me believe it could not accept a malformed representation. An exception would be thrown from `OffsetDateTime` when trying to use an invalid date, formatter o zone offset.

Not sure about the ISO date though, will look into it.

Reply:
Ah, nevermind, if it can only accept `DateTimeRfc1123` objects there is no need to test with ISO8601.